src/UI/Component/Input/Field/Factory.php

```diff
@@ -205,4 +206,60 @@ public function dependantGroup(array $inputs);
 	 * @return    \ILIAS\UI\Component\Input\Field\Checkbox
 	 */
 	public function checkbox($label, $byline = null);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     A TagInput is used to choose a subset amount of options out of a finite list
+	 *     of options. The TagInput is used whenever not all available options to choose from
+	 *     MUST or SHOULD be visible, e.g. because the amount is too high (such as all Usernames or a
+	 *     huge amount of Tags).
+	 *     By default, new Options whcih are not yet part of the List of given options, can't be
+	 *     submitted. This can be activated optionally (extendable).
+	 *   composition: >
+	 *     The Input is presented as a TextInput and prepended by already selected Options presented
+	 *     as tags with a Close-Button. The input is labeled by the $label given.
+	 *     Already selected Options are represented as Tags, a Tag contains the text-representation of
+	 *     the Option followed by a Close-button (e.g. [ Amsterdam X ] ).
+	 *     Suggested Options are listed in a Dropdown-List beneath the Textinput.
+	 *   effect: >
+	 *     As soon as the user types in the Textfield, the TagInput suggests matching Options from
+	 *     the a "local" or a "remote" list of Options (data-sources). Suggestions will appear after a defined
+	 *     amount of characters, one by default.
+	 *     Clicking on one of these options closes the List and transfers the selected option into
+	 *     the Input, displayed as a Tag with a Close-Button.
+	 *     By clicking on a Close-Button of a already selected Option, this Option will disappear
+	 *     from the Input.
+	 *     If no data-sources are provided, no suggestions will be provided.
```

Comment:
data-sources is a little technical, maybe replace with "If no options are offered".